Library/OcCryptoLib/Sha2.c

```diff
@@ -447,8 +451,13 @@ Sha512Update (
 
   ShiftedMsg = Data + RemLen;
 
-  Sha512Transform (Context, Context->Block, 1);
-  Sha512Transform (Context, ShiftedMsg, BlockNb);
+  if (mIsAvxEnabled) {
+    Sha512TransformAvx (Context->State, Context->Block, 1);
+    Sha512TransformAvx (Context->State, ShiftedMsg, BlockNb);
+  } else {
+    Sha512Transform (Context, Context->Block, 1);
+    Sha512Transform (Context, ShiftedMsg, BlockNb);
+  }
```

Comment:
Was there a strong reason for having the signatures of `Sha512Transform` and `Sha512TransformAvx` different? (e.g. can perhaps update signature of `Sha512Transform` to match new version - unless there is some requirement I'm not aware of for keeping it as is).
Suggesting this tentatively - happy to be overruled by @vit9696

Reply:
Thank you for the update